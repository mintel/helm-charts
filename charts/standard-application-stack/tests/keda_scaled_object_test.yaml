suite: Test Keda ScaledObjects
templates:
  - keda-scaled-object.yaml
release:
  namespace: test-namespace
tests:
  - it: Creates a ScaledObject for a Deployment
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      statefulset: false
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.scaleTargetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.scaleTargetRef.kind
          value: Deployment
      - equal:
          path: spec.scaleTargetRef.name
          value: example-app
      - equal:
          path: spec.fallback
          value:
            failureThreshold: 3
            replicas: 2

  - it: Creates a ScaledObject for a StatefulSet
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      statefulset: true
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.scaleTargetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.scaleTargetRef.kind
          value: StatefulSet
      - equal:
          path: spec.scaleTargetRef.name
          value: example-app
      - equal:
          path: spec.fallback
          value:
            failureThreshold: 3
            replicas: 2

  - it: Creates a ScaledObject for a custom scaleTargetRef
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.scaleTargetRef:
        apiVersion: apps/v2
        kind: CustomResource
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.scaleTargetRef.apiVersion
          value: apps/v2
      - equal:
          path: spec.scaleTargetRef.kind
          value: CustomResource
      - equal:
          path: spec.scaleTargetRef.name
          value: example-app
      - equal:
          path: spec.fallback
          value:
            failureThreshold: 3
            replicas: 2

  - it: Creates a ScaledObject with custom triggers
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.triggers.custom:
        - type: 'prometheus'
          metadata:
            metricName: 'http_requests_total'
            query: 'sum(rate(http_requests_total{service="podinfo-http"}[2m]))'
            serverAddress: 'http://prometheus-k8s.monitoring.svc:9090'
            threshold: '5'
        - type: 'prometheus'
          metadata:
            metricName: 'http_requests_total'
            query: 'sum(rate(http_requests_total{service="podinfo-grpc"}[5m]))'
            serverAddress: 'http://prometheus-k8s.monitoring.svc:9090'
            threshold: '10'
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.triggers
          value:
            - type: 'prometheus'
              metadata:
                metricName: 'http_requests_total'
                query: 'sum(rate(http_requests_total{service="podinfo-http"}[2m]))'
                serverAddress: 'http://prometheus-k8s.monitoring.svc:9090'
                threshold: '5'
            - type: 'prometheus'
              metadata:
                metricName: 'http_requests_total'
                query: 'sum(rate(http_requests_total{service="podinfo-grpc"}[5m]))'
                serverAddress: 'http://prometheus-k8s.monitoring.svc:9090'
                threshold: '10'

  - it: Creates a ScaledObject with fallback settings
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.fallback:
        failureThreshold: 1
        replicas: 10
        ignoredValue: true
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.fallback
          value:
            failureThreshold: 1
            replicas: 10

  - it: Creates a ScaledObject with advanced settings
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.advanced:
        restoreToOriginalReplicaCount: true
        horizontalPodAutoscalerConfig:
          name: test-hpa
          behavior:
            scaleDown:
              stabilizationWindowSeconds: 300
              policies:
              - type: Percent
                value: 100
                periodSeconds: 15
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.advanced
          value:
            restoreToOriginalReplicaCount: true
            horizontalPodAutoscalerConfig:
              name: test-hpa
              behavior:
                scaleDown:
                  stabilizationWindowSeconds: 300
                  policies:
                  - type: Percent
                    value: 100
                    periodSeconds: 15

  - it: Creates a ScaledObject with both targetCPUUtilizationPercentage and targetMemoryUtilizationPercentage triggers
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.triggers.targetCPUUtilizationPercentage: 100
      autoscaling.triggers.targetMemoryUtilizationPercentage: 50
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.triggers
          value:
            - type: cpu
              metricType: Utilization
              metadata:
                value: "100"
            - type: memory
              metricType: Utilization
              metadata:
                value: "50"

  - it: Creates a ScaledObject with targetCPUUtilizationPercentage trigger
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.triggers.targetCPUUtilizationPercentage: 100
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.triggers
          value:
            - type: cpu
              metricType: Utilization
              metadata:
                value: "100"

  - it: Creates a ScaledObject with targetMemoryUtilizationPercentage trigger
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.triggers.targetMemoryUtilizationPercentage: 50
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.triggers
          value:
            - type: memory
              metricType: Utilization
              metadata:
                value: "50"

  - it: Creates a ScaledObject with both targetCPUAverageValue trigger
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.triggers.targetCPUAverageValue: 101
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.triggers
          value:
            - type: cpu
              metricType: AverageValue
              metadata:
                value: "101"

  - it: Creates a ScaledObject with targetMemoryAverageValue trigger
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.triggers.targetMemoryAverageValue: 51
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.triggers
          value:
            - type: memory
              metricType: AverageValue
              metadata:
                value: "51"

  - it: Creates a ScaledObject with both targetCPUAverageValue and targetMemoryAverageValue triggers
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.triggers.targetCPUAverageValue: 101
      autoscaling.triggers.targetMemoryAverageValue: 51
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.triggers
          value:
            - type: cpu
              metricType: AverageValue
              metadata:
                value: "101"
            - type: memory
              metricType: AverageValue
              metadata:
                value: "51"

  - it: Check utilization takes precedence over average value when both specified
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.triggers.targetCPUUtilizationPercentage: 100
      autoscaling.triggers.targetMemoryUtilizationPercentage: 50
      autoscaling.triggers.targetCPUAverageValue: 101
      autoscaling.triggers.targetMemoryAverageValue: 51
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.triggers
          value:
            - type: cpu
              metricType: Utilization
              metadata:
                value: "100"
            - type: memory
              metricType: Utilization
              metadata:
                value: "50"


suite: Test Keda ScaledObject Constraints
templates:
  - keda-scaled-object.yaml
release:
  namespace: test-namespace
tests:
  - it: Creates a ScaledObject with non-constrained settings
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.minReplicaCount: 5
      autoscaling.maxReplicaCount: 10
      autoscaling.cooldownPeriod: 500
      autoscaling.pollingInterval: 60
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.scaleTargetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.scaleTargetRef.kind
          value: Deployment
      - equal:
          path: spec.scaleTargetRef.name
          value: example-app
      - equal:
          path: spec.maxReplicaCount
          value: 10
      - equal:
          path: spec.minReplicaCount
          value: 5
      - equal:
          path: spec.pollingInterval
          value: 60
      - equal:
          path: spec.cooldownPeriod
          value: 500 

  - it: Creates a ScaledObject with constrained settings
    set:
      global.clusterEnv: qa
      autoscaling.enabled: true
      autoscaling.minReplicaCount: 0
      autoscaling.maxReplicaCount: 500
      autoscaling.cooldownPeriod: 5
      autoscaling.pollingInterval: 5
    asserts:
      - isKind:
          of: ScaledObject
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.name
          value: example-app
      - equal:
          path: spec.scaleTargetRef.apiVersion
          value: apps/v1
      - equal:
          path: spec.scaleTargetRef.kind
          value: Deployment
      - equal:
          path: spec.scaleTargetRef.name
          value: example-app
      - equal:
          path: spec.maxReplicaCount
          value: 10
      - equal:
          path: spec.minReplicaCount
          value: 1
      - equal:
          path: spec.pollingInterval
          value: 10
      - equal:
          path: spec.cooldownPeriod
          value: 60
