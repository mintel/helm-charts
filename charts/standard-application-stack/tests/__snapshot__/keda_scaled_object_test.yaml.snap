Check utilization takes precedence over average value when both specified:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
      triggers:
        - metadata:
            value: "100"
          metricType: Utilization
          type: cpu
        - metadata:
            value: "50"
          metricType: Utilization
          type: memory
Creates a ScaledObject for a Deployment:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
Creates a ScaledObject for a StatefulSet:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: StatefulSet
        name: example-app
Creates a ScaledObject for a custom scaleTargetRef:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v2
        kind: CustomResource
        name: example-app
Creates a ScaledObject with advanced settings:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced:
        horizontalPodAutoscalerConfig:
          behavior:
            scaleDown:
              policies:
                - periodSeconds: 15
                  type: Percent
                  value: 100
              stabilizationWindowSeconds: 300
          name: test-hpa
        restoreToOriginalReplicaCount: true
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
Creates a ScaledObject with both targetCPUAverageValue and targetMemoryAverageValue triggers:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
      triggers:
        - metadata:
            value: "101"
          metricType: AverageValue
          type: cpu
        - metadata:
            value: "51"
          metricType: AverageValue
          type: memory
Creates a ScaledObject with both targetCPUAverageValue trigger:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
      triggers:
        - metadata:
            value: "101"
          metricType: AverageValue
          type: cpu
Creates a ScaledObject with both targetCPUUtilizationPercentage and targetMemoryUtilizationPercentage triggers:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
      triggers:
        - metadata:
            value: "100"
          metricType: Utilization
          type: cpu
        - metadata:
            value: "50"
          metricType: Utilization
          type: memory
Creates a ScaledObject with custom triggers:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
      triggers:
        - metadata:
            metricName: http_requests_total
            query: sum(rate(http_requests_total{service="podinfo-http"}[2m]))
            serverAddress: http://prometheus-k8s.monitoring.svc:9090
            threshold: "5"
          type: prometheus
        - metadata:
            metricName: http_requests_total
            query: sum(rate(http_requests_total{service="podinfo-grpc"}[5m]))
            serverAddress: http://prometheus-k8s.monitoring.svc:9090
            threshold: "10"
          type: prometheus
Creates a ScaledObject with default fallback based on minReplicaCount:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 1
        replicas: 3
      maxReplicaCount: 5
      minReplicaCount: 3
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
Creates a ScaledObject with explicit fallback settings:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 1
        replicas: 10
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
Creates a ScaledObject with targetCPUUtilizationPercentage trigger:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
      triggers:
        - metadata:
            value: "100"
          metricType: Utilization
          type: cpu
Creates a ScaledObject with targetMemoryAverageValue trigger:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
      triggers:
        - metadata:
            value: "51"
          metricType: AverageValue
          type: memory
Creates a ScaledObject with targetMemoryUtilizationPercentage trigger:
  1: |
    apiVersion: keda.sh/v1alpha1
    kind: ScaledObject
    metadata:
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      advanced: {}
      cooldownPeriod: 300
      fallback:
        failureThreshold: 3
        replicas: 2
      maxReplicaCount: 5
      minReplicaCount: 2
      pollingInterval: 30
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: example-app
      triggers:
        - metadata:
            value: "50"
          metricType: Utilization
          type: memory
